proposals/ytt/004-schema-validation/README.md

```diff
@@ -0,0 +1,1523 @@
+> - [ ] describe the fallback strategy where authors can represent their rule where there _is_ no OpenAPI equivalent.
+
+When the user inspects the schema:
+
+```
+$ ytt ... --data-values-schema-inspect -o openapi-v3
+```
+
+Validations defined on schema are included in that export.
+
+
+OpenAPI v3 / JSON Schema validation keywords:
+
+Supported (ytt rule ==> OpenAPI v3 / JSON Schema property):
+- `max=` / `@ytt:assert.max()` ==> [maximum](https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#section-5.2)
+- `min=` / `@ytt:assert.min()` ==> [minimum](https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#section-5.4)
+- `max_len=` / `@ytt:assert.max_len()`:
+  - type: `string` ==> [maxLength](https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#section-5.6)
+  - type: `array` ==> [maxItems](https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#section-5.10)
+  - type: `map` ==> [maxProperties](https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#section-5.13)
+- `min_len=` / `@ytt:assert.min_len()`:
+  - type: `string` ==> [minLength](https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#section-5.7)
+  - type: `array` ==> [minItems](https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#section-5.11)
+  - type: `map` ==> [minProperties](https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#section-5.14)
+- `one_of=` / `@ytt:assert.one_of()`: [enum](https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#section-5.20)
+- `format=` / `@ytt:assert.format()` : [format](https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#section-7) \
+   Format names are identical:
+  - [date-time](https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#section-7.3.1)
+  - [email](https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#section-7.3.2)
+  - [hostname](https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#section-7.3.3)
+  - [ipv4](https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#section-7.3.4)
+  - [ipv6](https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#section-7.3.5)
+  - [uri](https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#section-7.3.6)
+  - [uriref](https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#section-7.3.7)
+
+The following are deferred:
+
+- uniqueItems
+- exclusiveMaximum (is this useful?)
+- exclusiveMinimum (is this useful?)
+- [multipleOf](https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#section-5.1)
+- pattern (This string SHOULD be a valid regular expression, according to the ECMA 262 regular - expression dialect)
```

Comment:
@cppforlife noted:
[ECMA 262](https://262.ecma-international.org/12.0/#sec-patterns) != [RE2](https://github.com/google/re2/wiki/Syntax)

Q: what approach have others who have implemented this feature (specifically JSON Schema validation keyword `pattern`) in Go done?